homeworkLecturesW10D03/src/ba/bitcamp/Files/MyPanel.java

```diff
@@ -0,0 +1,181 @@
+
+/**
+ * @author Gordan
+ *
+ *         Making MyPanel that has next specifications: JTextArea used for
+ *         writing some text JMenuBar that contains JManu with tree JMenuItems:
+ *         New,Open and Save As JScrollPane with horizontal and vertical
+ *         scrollbars
+ *
+ *         Every JMenuItem has a specific ActionListener: New creates new empty
+ *         text area Open is used to open choosed text file Save As is used to
+ *         save text file at choosed location and under choosed name
+ */
+public class MyPanel extends JPanel {
+
+	private static final long serialVersionUID = -3466550005830151593L;
+
+	// Creating private objects used for developing MyPanel object
+
+	// Creating text area
+	private JTextArea textArea = new JTextArea();
+
+	// Creating menu
+	private JMenu menu = new JMenu("File");
+
+	// Creating menu bar
+	private JMenuBar menuBar = new JMenuBar();
+
+	// Creating menu items
+	private JMenuItem newFile = new JMenuItem("New...");
+	private JMenuItem open = new JMenuItem("Open...");
+	private JMenuItem save = new JMenuItem("Save As");
+
+	// Creating scroll pane
+	private JScrollPane scrollPane = new JScrollPane(textArea,
+			JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED,
+			JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+
+	/**
+	 * Constructor for creating MyPanel object with text area, menu bar and
+	 * scroll pane
+	 */
+	public MyPanel() {
+
+		// Seting panel's layout
+		setLayout(new BorderLayout());
+
+		// Adding menu bar to the north of panel
+		add(menuBar, BorderLayout.NORTH);
+
+		// Adding menu to the menu bar
+		menuBar.add(menu);
+
+		// Adding menu items to the menu
+		menu.add(newFile);
+		menu.add(open);
+		menu.add(save);
+
+		// Making text area
+		textArea.setLineWrap(true);
+
+		// Adding scroll pane to the center of the panel
+		add(scrollPane, BorderLayout.CENTER);
+
+		/*
+		 * ActionListener for creating empty text file
+		 */
+		newFile.addActionListener(new ActionListener() {
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				textArea.setText("");
+			}
+		});
+
+		/*
+		 * ActionListener for opening existing text file
+		 */
+		open.addActionListener(new ActionListener() {
+
+			private BufferedReader br;
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				try {
+					UIManager.setLookAndFeel(UIManager
+							.getSystemLookAndFeelClassName());
+				} catch (ClassNotFoundException | InstantiationException
+						| IllegalAccessException
+						| UnsupportedLookAndFeelException e1) {
+					e1.printStackTrace();
+				}
+				JFileChooser openFile = new JFileChooser();
+				int action = openFile.showOpenDialog(open);
+				if (action == JFileChooser.OPEN_DIALOG) {
+					File file = openFile.getSelectedFile();
+					try {
+						FileReader fr = new FileReader(file);
+						br = new BufferedReader(fr);
+						String line = br.readLine();
+						String text = line + "\n";
+						while (line != null) {
+							line = br.readLine();
```

Comment:
Concatenating strings in a loop is very ineffiecient since, each time string is concatenated, new string instance is created.

Instead i suggest using `StringBuilder`:

```
StringBuilder fileData = new StringBuilder();
String line = null;

while ((line = fileReader.readLine()) != null) {
fileData.append(line).append('\n');
}

```